Error when importing a workflow with exotic characters

Importing a workflow with the following variable named `DOCUMENT_JSON` and its value:
```json
{      "documents": [          {              "id": "1",              "text": "This document is in English."          },          {              "id": "2",              "text": "Este documento estÃ¡ en inglÃ©s."          },          {              "id": "3",              "text": "Ce document est en anglais."          },          {              "id": "4",              "text": "æ¬æä»¶ä¸ºè±æ"          },                          {              "id": "5",              "text": "Ð­ÑÐ¾Ñ Ð´Ð¾ÐºÑÐ¼ÐµÐ½Ñ Ð½Ð°ÑÐ¾Ð´Ð¸ÑÑÑ Ð½Ð° Ð°Ð½Ð³Ð»Ð¸Ð¹ÑÐºÐ¾Ð¼ ÑÐ·ÑÐºÐµ."          }      ]  }
```

fails because of the weird characters (scroll to the far right of the value) with the following exception:

```
[2018-08-22 09:35:13,858 461159-175 WARN    o.h.e.j.s.SqlExceptionHelper] SQL Error: 1366, SQLState: 22007                                                                                              [1195/5212]
[2018-08-22 09:35:13,858 461159-175 ERROR   o.h.e.j.s.SqlExceptionHelper] (conn=575) Incorrect string value: '\xC2\x9C\xC2\xAC\xC3\xA6...' for column 'PA_VALUE' at row 1                                          
[2018-08-22 09:35:13,873 461159-175 WARN  r.e.ExceptionHandlerController] Exception: could not execute statement; SQL [n/a]; nested exception is org.hibernate.exception.DataException: could not execute statement
[2018-08-22 09:35:13,876 461159-175 WARN          o.e.j.s.ServletHandler]                                                                                                                                          
org.springframework.web.util.NestedServletException: Request processing failed; nested exception is org.springframework.dao.DataIntegrityViolationException: could not execute statement; SQL [n/a]; nested excepti
on is org.hibernate.exception.DataException: could not execute statement                                                                                                                                           
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:979)                                                                                                              
        at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:869)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:843)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:812)
...
Caused by: java.sql.SQLException: Incorrect string value: '\xC2\x9C\xC2\xAC\xC3\xA6...' for column 'PA_VALUE' at row 1
Query is: insert into METADATA_KEY_VALUE (CATALOGOBJECTREVISION, PA_KEY, LABEL, PA_VALUE, ID) values (?, ?, ?, ?, ?), parameters [70,'DOCUMENTS_JSON','variable','{      "documents": [          {              "id
": "1",              "text": "This document is in English."          },          {              "id": "2",              "text": "Este documento estÃ¡ en inglÃ©s."          },          {              "id": "3",
            "text": "Ce document est en anglais."          },          {              "id": "4",              "text": "æ¬æä»¶ä¸ºè±æ"          }Ã¡        Ã©               {              "id": "5",
    "text": "Ð­ÑÐ¾Ñ Ð´Ð¾ÐºÑÐ¼ÐµÐ½Ñ Ð½Ð°Ñ                                                                           æ¬æä»¶ä¸ºè±æ
Ð¾Ð´Ð¸ÑÑÑ ÐÐ­ÑÐ¾ÑÐÐ´Ð¾ÐºÑÐ¼ÐµÐ½ÑÑÐ½Ð°ÑºÐµ."          }      ]  }',573]
Ð¾Ð´Ð¸ÑÑÑoÐ½Ð°aÐ°Ð½Ð³Ð»Ð¸Ð¹ÑÐºÐ¾Ð¼uÑÐ·ÑÐºÐµeryTool.exceptionWithQuery(LogQueryTool.java:153)
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(AbstractQueryProtocol.java:255)
        at org.mariadb.jdbc.MariaDbPreparedStatementClient.executeInternal(MariaDbPreparedStatementClient.java:209)
        ... 73 more
```

--- a/src/main/java/org/ow2/proactive/catalog/util/ProActiveMySQL5InnoDBDialect.java
+++ b/src/main/java/org/ow2/proactive/catalog/util/ProActiveMySQL5InnoDBDialect.java
@@ -1,0 +1,42 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.catalog.util;
+
+import org.hibernate.dialect.MySQL5InnoDBDialect;
+
+
+/**
+ * This class is to be used with MySQL or MariaDB to allow full utf-8 support.
+ *
+ * @author ActiveEon Team
+ * @since 27/08/18
+ */
+public class ProActiveMySQL5InnoDBDialect extends MySQL5InnoDBDialect {
+    @Override
+    public String getTableTypeString() {
+        return " ENGINE=InnoDB DEFAULT CHARSET=utf8mb4";
+    }
+}

--- a/src/main/resources/application.properties
+++ b/src/main/resources/application.properties
@@ -8,9 +8,20 @@ logging.level.org.springframework.web=info
 server.compression.enabled=true
 server.contextPath=/
 
-# DataSource settings: set here your own configurations for the database connection.
+##############
+# DATASOURCE #
+##############
+
+# The default settings are using hsqldb
 #spring.datasource.driverClassName=org.hsqldb.jdbc.JDBCDriver
-#spring.datasource.url=jdbc:hsqldb:file:/tmp/proactive/catalogObject-catalog;create=true;hsqldb.tx=mvcc;hsqldb.applog=1;hsqldb.sqllog=0;hsqldb.write_delay=false
+#spring.datasource.url=jdbc:hsqldb:file:/tmp/proactive/catalog;create=true;hsqldb.tx=mvcc;hsqldb.applog=1;hsqldb.sqllog=0;hsqldb.write_delay=false
+
+# For MariaDB/MySQL use the following settings
+# note the ProActiveMySQL5InnoDBDialect class that enforces the utf8mb4 charset
+#spring.datasource.driverClassName=org.mariadb.jdbc.Driver
+#spring.jpa.database-platform=org.ow2.proactive.catalog.util.ProActiveMySQL5InnoDBDialect
+#spring.datasource.url=jdbc:mariadb://localhost:3306/catalog
+
 #spring.datasource.username=root
 #spring.datasource.password=
 

